### PR TITLE
drivers/at24cxxx: MTD wrapper for at24cxxx

### PR DESCRIPTION
**Pull request by fabian18**
### Contribution description
Analog to the MTD wrapper for `at25xxx` EEPROMs, this contribution consists of an MTD wrapper for `at24cxxx` EEPROMs.

### Testing procedure
There is a test: `tests/mtd_at24cxxx` which is basically a copy of `tests/mtd_at25xxx`.

### Issues/PRs references

 #13877

### CODE REVIEW

File: drivers/at24cxxx/mtd/mtd.c [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    mtd->pages_per_sector = 1;
+    mtd->sector_count = DEV(mtd)->params.eeprom_size /
+                        DEV(mtd)->params.page_size;
+    return 0;
+}
+
+static int _mtd_at24cxxx_read(mtd_dev_t *mtd, void *dest, uint32_t addr,
+                              uint32_t size)
+{
+    return at24cxxx_read(DEV(mtd), addr, dest, size) == AT24CXXX_OK ? 0 : -EIO;
```
Comment (benpicco) [inline drivers/at24cxxx/mtd/mtd.c:50]: You could just make the `at24cxxx` functions return 0 or `-EIO` on their own.

Comment (fabian18) [reply drivers/at24cxxx/mtd/mtd.c:50]: Are you sure this makes a difference?
```
at24cxxx_read(...) {
...
    return check == AT24CXXX_OK ? check : -EIO;
}
```
It just propagates the value checking to a lower level, while the requirement to return `-EIO` exists on `MTD` level.

Comment (benpicco) [reply drivers/at24cxxx/mtd/mtd.c:50]: I was hoping clever LTO could throw out the wrapper functions if we just pass on the return values, but I did not try this.

Comment (benpicco) [reply drivers/at24cxxx/mtd/mtd.c:50]: `at24cxxx_read()` will either return 0 or `-ERANGE`, so you could just 

    return at24cxxx_read(DEV(mtd), addr, dest, size);

here and comply with the API.

Comment (fabian18) [reply drivers/at24cxxx/mtd/mtd.c:50]: `at24cxxx_read()` will pass through errors from `i2c_read_regs()`.
Of course I´d rather just like to do the function call.
But what if someone relies on `-EIO` explicitly? Altough, I did not find it in existing code.
If you say that´s not a problem for the benefit of  LTO, then I´ll change.

Comment (benpicco) [reply drivers/at24cxxx/mtd/mtd.c:50]: tbh I was also hoping to get some consistency that way, e.g. that the user can expect the same error values from MTD and the device itself.

right now `mtd_read()` will return

> `-EOVERFLOW if addr or count are not valid, i.e. outside memory`

and `at24cxxx_read()` returns 

> `-ERANGE if pos + len is out of bounds`

Maybe there is a reason for that, but at a glance this looks inconsistent.

**Comment by benpicco** [comment]
I added a `page_write()` function in #14038 (untested)

**Comment by fabian18** [comment]
Thanks for merging.

> I added a page_write() function in #14038 (untested)

```C
    uint8_t expected_write_data[] = WRITE_CHARACTERS;
    uint32_t page = WRITE_POSITION / at24cxxx_dev.params.page_size;
    uint32_t offset = WRITE_POSITION - (page * at24cxxx_dev.params.page_size);
    size_t remaining = at24cxxx_dev.params.page_size -
                       (WRITE_POSITION % at24cxxx_dev.params.page_size);

    check = at24cxxx_write_page(&at24cxxx_dev, page, offset,
                                expected_write_data,
                                sizeof(expected_write_data));

    if (check != (int)MIN(sizeof(expected_write_data), remaining)) {
        printf("[FAILURE] at24cxxx_write_page: %d (remaining size = %zu)\n",
               check, remaining);
        return 1;
    }
    else {
        puts("[SUCCESS] at24cxxx_write_page");
    }
```

```
2020-06-25 17:48:25,433 # main(): This is RIOT! (Version: 2020.07-devel-1530-g30b20-mtd_pagewise)
2020-06-25 17:48:25,437 # Starting tests for module at24cxxx
2020-06-25 17:48:25,439 # EEPROM size: 32768 byte
2020-06-25 17:48:25,440 # Page size  : 64 byte
2020-06-25 17:48:25,443 # [SUCCESS] at24cxxx_init
2020-06-25 17:48:30,567 # [SUCCESS] at24cxxx_erase
2020-06-25 17:48:30,570 # [SUCCESS] at24cxxx_write_byte
2020-06-25 17:48:30,573 # [SUCCESS] at24cxxx_read_byte
2020-06-25 17:48:30,575 # [SUCCESS] write_byte/read_byte
2020-06-25 17:48:30,579 # [SUCCESS] at24cxxx_write_page
2020-06-25 17:48:30,584 # [SUCCESS] at24cxxx_write
2020-06-25 17:48:30,588 # [SUCCESS] at24cxxx_read
2020-06-25 17:48:30,589 # [SUCCESS] write/read
2020-06-25 17:48:30,596 # [SUCCESS] at24cxxx_set
2020-06-25 17:48:30,600 # [SUCCESS] set/read
2020-06-25 17:48:30,603 # Finished tests for module at24cxxx
```
`at24cxxx_write_page()` seems to work fine.

**Pull request merged**